### ISSUE TITLE
Core: mupen64plus Rice video plugin won't display properly; float values are displayed as integers

### ISSUE THREAD

**Issue by leodexe** [TASVideos/BizHawk#2064]
[//]: # "This issue description supports Markdown syntax (this is what comments look like). There's a cheatsheet here: https://guides.github.com/features/mastering-markdown/"
[//]: # "You can leave these comments here or delete them. Also, please remember to search for similar issues before writing anything, including in closed issues!"
[//]: # "One more thing: if you're on Linux, please comment on #1430 instead of opening an issue so we don't annoy the other devs."

### Summary
https://youtu.be/eVz6bTggyEI
Well you can see the problem in the video, for whatever reason the textures sometimes will not display properly when the game zooms in and zooms out of the view. Selecting any video plugin other than Rice will not even boot the game at all (crashes after N64 logo).
Also the RAM Search does not convert float values properly, if you try searching a float value (i.e Luigi's X-Air Maximum Speed which is 22.5, it will show in the results as an integer value 225, or even worse it won't show nothing) https://nickthename.github.io/ssbframedata/images/charpropertieschart.jpg
As a reference you can try searching any of these float values in this chart, at least for me it shows floats as integers which is a bad thing, also poking those float values will actually convert them into integers, breaking most of the RAM Search functionality as it can't do what's supposed to do.
### Repro
[//]: # "If you can't figure out the list of steps, delete this section and put 'heisenbug' in the summary somewhere. If a Lua script can cause the bug, you can embed that instead (as simple as possible please)."
1. To reproduce the video issue, you need to get your hands on this ROM Hack
https://github.com/JSsixtyfour/smashremix/releases
Instructions to build the ROM are easy, you only need to legally acquire a copy of Super Smash Bros. somewhere.
2. Load Smash Remix.
3. If the game doesn't boot at all, try Rice Video Plugin as it is the only one that worked for me.
4. Select any character, navigate to the Stage Selection Screen and pick New Pork City. This is the stage where the video issue pops in more often, other stage may have different display issues.
5. After a stage has been loaded and the game begins, regardless if it is training mode, 1p mode or versus mode, try using the RAM Search function.
6. Search some of the float values provided in the character properties chart, it should be any of the characters you picked.
7. Try poking any float value, it should now convert into a integer value.

### Host env.
[//]: # "List the computers you've found the bug with. If there's a version that doesn't have the bug, please put that in too. Here are some examples:"
- BizHawk 2.4.2, Win10Pro Build 1909 x64 bits, Motherboard H77H2-EM, intel Core i5-3450.
About the float value, this video https://www.youtube.com/watch?v=4YET7nWCglk
this youtuber "Mayyro" uses an older version of Bizhawk, 1.11.7 to be specific, in what appears to be in Windows 7, there it correctly displays, search and convert float values properly. I also tested that older version of BizHawk on my Windows 10 PC and yet the RAM Search is still broken. HOWEVER, that version includes Jabo Graphic Plugin 1.6.1, which correctly displays the same stage this version of Rice plugin bundled with BizHawk 2.4.2 cannot, also tested the Rice Video, Glide64 and Glide64mk2 on version 1.11.7 of BizHawk, in current Bizhawk 2.4.2, only the Rice video plugin works with some artifacts, but in 1.11.7 glide64 works with minor visual artifacts which is better than not working at all in version 2.4.2, that said, this current version of Bizhawk should include this version of Jabo's plugin which is old yet stable and more functional than any provided with the current build. Or even Jabo's 1.7 which works better than 1.6 and is less outdated. Also it is worth mentioning that Bizhawk 1.11.7 had support for Dynamic Recompiler core type, which current Bizhawk 2.4.2 only supports interpreter and pure interpreter core types, excluding Dynamic Recompiler.

[//]: # "(screenshots, if applicable)"
![imagen](https://user-images.githubusercontent.com/24327014/83089382-2ec41e80-a064-11ea-9d48-15225672f7a3.png)
![imagen](https://user-images.githubusercontent.com/24327014/83089400-3b487700-a064-11ea-98b8-da17efe76657.png)
![imagen](https://user-images.githubusercontent.com/24327014/83089442-587d4580-a064-11ea-8bea-6c0dbb2fa762.png)
![imagen](https://user-images.githubusercontent.com/24327014/83089454-616e1700-a064-11ea-80db-2320a3a00a2f.png)
![imagen](https://user-images.githubusercontent.com/24327014/83089473-68952500-a064-11ea-8a71-a5050d7d287d.png)
![imagen](https://user-images.githubusercontent.com/24327014/83089505-721e8d00-a064-11ea-906c-cdf2ac7bfcf1.png)
![imagen](https://user-images.githubusercontent.com/24327014/83099093-50300500-a07a-11ea-80f6-0c16f28eed4f.png)
![imagen](https://user-images.githubusercontent.com/24327014/83098704-55408480-a079-11ea-81f1-1c4e25899e7e.png)
![imagen](https://user-images.githubusercontent.com/24327014/83098804-920c7b80-a079-11ea-93c5-787d5da7a7c6.png)

That said, hope this helps with your build.

**Comment by YoshiRulz** [comment]
Do you know if this warning:
> The original xdelta will generate a smash rom that is compatible with our ASM code. Much of our edits are done within the compressed files within the rom. If you utilize a vanilla Smash 64 rom, it will not work correctly.

...from the hack's readme applies to releases, or only when building from source?

It'll probably be easier to get this romhack working with GLideN64 than to fix the old Rice plugin (or to add Jabo, given it works on upstream.

---

Floats in Ram Search not working might be unrelated, I'll look into that first.

**Comment by leodexe** [comment]
Yes, my computer lacks a dedicated graphics card. I will try with my GTX 750 Ti, I am not using it since I don't really need it. I will post results later.

So far I have tested Bizhawk 1.17 and 2.42 on another H77 Motherboard with intel Core i7-3770, which also uses Windows 10 Pro Build 1909 x64, with intel HD Graphics 4000 and it has the same issues I am having with my H77 Motherboard with intel Core i5-3450 and intel HD Graphics 2500.

However, I also tested both Bizhawk versions on another setup which managed to work properly, displaying graphics correctly and the RAM Search function correctly convert and show float values.
This functional setup is a H61 Motherboard with intel Core i3-2120 and Windows 8.1 Pro x64.

Also Mayyro's setup where he used Bizhawk 1.17 with Windows 7 works correctly, which may suggest that Windows 10 is the source of the problem of the RAM Search function being broken.

About the RAM Search function, on what settings did you use? What Windows version do you use?

**Comment by YoshiRulz** [comment]
That's really helpful data to have, thank you. I deleted my `config.ini` on 2.4.2 to get the defaults, and my Windows machine is Intel/NVIDIA with Win10 Home 2004 (Insider preview build 19551.1005).

**Comment by leodexe** [comment]
Could you add support for Jabo's plugin on the next release please? Or if you can't could you at least me point to the latest version of Bizhawk that supports Jabo's Graphic Plugin for and (optionally) Dynamic Recompiler Core Type for the N64?

**Comment by YoshiRulz** [comment]
That'd be [1.13.2](https://github.com/TASVideos/BizHawk/releases/tag/1.13.2).

**Comment by leodexe** [comment]
I don't know if it's worth mentioning this but I have tested older BizHawk versions with my core i5-3450 Ivy Bridge CPU (no graphic cards yet) and I had the following results:
Tested on default settings, with N64 ROM Hack Smash Remix 0.9.3b

Bizhawk v1.11.7 to v1.13.2
Jabo plugin: works.
Glide64 plugin: works.
GlideN64 plugin: works.
Glide64mk2 plugin: it works with major slowdowns every time the game has to load a new texture for the first time, after all the textures has been loaded the slowdown disappears. 
Rice plugin: works with the issue described in #2064 

Bizhawk v2.0 to v2.2
Glide64 plugin: doesn't work. (after the N64 logo it will just display garbage)
GlideN64 plugin: works.
Glide64mk2 plugin: doesn't work. It shows this error message every time I try to load the game with this plugin.
![glide64mk2erroronbizhawk2 2](https://user-images.githubusercontent.com/24327014/83319582-ce70e080-a20d-11ea-9366-3a183bd037a3.png)
Rice plugin: works with the issue described in #2064 

Bizhawk v2.2.1 to v2.3
The N64 Core crashes after selecting the ROM, it won't even boot to the N64 Logo, the program just closes itself without displaying any error and no config.ini is saved regardless of how many settings are modified.

Bizhawk v2.3.1 to v2.4.2
Glide64 plugin: doesn't work. (after the N64 logo it will just display garbage)
GlideN64 plugin: doesn't work, same problem as Glide64.
Glide64mk2 plugin: works.
Rice plugin: works with the issue described in #2064
